### PR TITLE
Add i18n for menu headings and commands

### PR DESCRIPTION
**Pull request by gtritchie**
### Intent

This PR enables i18n for the menu headings and command headings/descriptions that are defined in the `Commands.cmd.xml` file.  Shortcuts in that file have not been addressed here.

Only English is generated at this time, so no end-user visible changes should be noticed.

> This was originally submitted as an external PR here: https://github.com/rstudio/rstudio/pull/9507
> 
> Getting it staged for merge was proving tricky, so agreed to take ownership of final tweaks and moved the changes onto this local branch instead of a fork.

### GWT Dev Workflow Changes (IMPORTANT: IDE DEVS MUST READ)

If you have not modified `Commands.cmd.xml` then nothing changes. You can build GWT using any of the usual `ant` commands (`draft`, `devmode`, `desktop`, or just plain `ant`).

If you change `Commands.cmd.xml` in any way, including whitespace, then `ant` will trigger the localization scripts and regenerate the support files. These modified files must be committed with modified `Command.cmd.xml`. Failure to do this will break the build.

The quickest way to ensure you've done this is `ant generate-i18n`. This will ensure the loc files match the current Commands.cmd.xml without the overhead of a full GWT build.

The localization scripts are written in Python (v3 required), and require some packages. These are specified in `rstudio/src/gwt/tools/i18n-helpers/commands.cmd.xml/requirements.txt`. These can be installed using `pip install -r requirements.txt`.

On Windows, I had to follow steps here to install the requirements (i.e. I had to download and install prebuilt binary): https://stackoverflow.com/questions/33785755/getting-could-not-find-function-xmlcheckversion-in-library-libxml2-is-libxml2

### Implementation Details

The need to run the localization scripts is gated on an MD5 checksum of `Commands.cmd.xml`. It is generated by `ant` and stored in `Commands.cmd.xml.MD5`. If the checksum matches, then the localization scripts will not execute.

If working on a Windows machine, the Commands.cmd.xml file will be checked out with Linux LF line-endings, not CR/LF. Otherwise the checksum won't match. This is configured via a .gitattributes file included in this PR.

#### Full Implementation Details

See original notes in https://github.com/rstudio/rstudio/pull/9507 for implementation details, as well as how to try out the localization process.

### Automated Tests

No automated tests were added.

### QA Notes

For now, just verify that the English UI is functioning normally (menus should look the same as before).

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

**Comment by gtritchie** [comment]
Thought I had it working, but still hitting problems with the checksum mechanism. Moving back to draft. Sigh.

**Comment by gtritchie** [comment]
Moving back to ready to review. Looks like the checksum wasn't being updated due to timestamp issues. Should now be fixed, but I'm running test builds on all platforms to double-check.

File: src/gwt/src/org/rstudio/core/rebind/command/CommandBundleGenerator.java
Comment (melissa-barca) [inline src/gwt/src/org/rstudio/core/rebind/command/CommandBundleGenerator.java:325]:
```suggestion
```
Should this line be removed since you added +326?

File: src/gwt/src/org/rstudio/core/rebind/command/CommandBundleGenerator.java
Comment (melissa-barca) [inline src/gwt/src/org/rstudio/core/rebind/command/CommandBundleGenerator.java:383]:
```suggestion
      }
      else
      {
```
nit: formatting

**Review by melissa-barca — COMMENTED** [review]
This is great. Thank you again @ca-scribner & @skyeturriff to getting this started.

File: src/gwt/tools/i18n-helpers/commands.cmd.xml/i18n_gwt_interface.py
Comment (jmcphers) [inline src/gwt/tools/i18n-helpers/commands.cmd.xml/i18n_gwt_interface.py:90]: Does this need to be finished before merge?

**Review by jmcphers — APPROVED** [review]
LGTM pending comments above.

**Review by melissa-barca — APPROVED** [review]
LGTM!

**Comment by gtritchie** [comment]
Local Selenium testing shows some menu-related identifiers have changed as a result of this PR; will need to investigate that further before I merge this. Might be we just change the automation to match, but would be better if the identifiers stayed the same.

**Comment by gtritchie** [comment]
Internal note: seeing various issues with menu labels. For example, notice lack of a space between words in some cases, or a space shown as an underscore in others:

Screen Shot 2021-11-03 at 4 30 06 PM

**Comment by gtritchie** [comment]
Back to draft. :-(

**Comment by gtritchie** [comment]
Also some quirks on Windows desktop, for example see multiple accelerators (underlines) on Tools / Version Control:

underscore-problems

**Comment by skyeturriff** [comment]
> @skyeturriff Do you see this on your end? Wondering if this is something I introduced.

Hey @gtritchie ! I'm having some issues with my local dev environment, but once I get it fixed I will run from our fork and confirm!

**Comment by gtritchie** [comment]
> Hey @gtritchie ! I'm having some issues with my local dev environment atm, but once I get it fixed I will run from our fork and confirm!

I'm also debugging it, I don't think I introduced this problem, so will require some tweaks to the Python scripts which I can work on (unless you've already started, let me know!). REALLY want to get this in for you; was very close to merging then noticed this.

**Comment by gtritchie** [comment]
@skyeturriff Ok, I think I've got it fixed via d7e969e60c0e3d4bb51ba2583a6675329a49bc6d. Going to re-run some tests, and if that goes well should be able to merge this today (fingers crossed).

**Pull request merged**
